service/actions/channels.js

```diff
@@ -381,6 +381,11 @@ export function deleteChannel(teamId, channelId) {
             return;
         }
 
+        if (beforeDelete) {
+            const {action, args, _dispatch, _state} = beforeDelete;
```

Comment:
as far as I can tell `beforeDelete` is a boolean as of the parameters in here https://github.com/mattermost/mattermost-mobile/pull/222/files#diff-21a00938e5c0137fd9703232f07fa851R130

The other thing that I might point out here is that this PR https://github.com/mattermost/mattermost-mobile/pull/221 has an Event Emitter that could be used here to emit the event and it can be subscribe elsewhere to take the action